### PR TITLE
ILLiad Setup

### PR DESCRIPTION
**Pull request by erinesullivan**
# Overview
We need to apply ILLiad's API to My Account. This pull request creates the class, environment variables, and stub for testing.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
       NELNET_PAYMENT_URL: 'http://localhost:4444'
       JWT_SECRET: 'myjwtsecret'
       PATRON_ACCOUNT_BASE_URL: 'http://localhost:4567'
+      ILLIAD_API_KEY: 'YOURAPIKEY'
```
Comment (niquerio) [inline .github/workflows/tests.yml:16]: The ILLiad key and host needed to be added to github actions test.yml. Sadly I can't just import a file with all of the keys to make testing work. :crying_cat_face:

**Comment by niquerio** [comment]
I'm going to make sure the ILLiad Keys are in kubernetes before merging.

File: models/illiad_client.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  def initialize()
+    self.class.headers 'ApiKey' => ENV.fetch('ILLIAD_API_KEY')
+    self.class.headers 'Accept' => 'application/json'
+  end
+
+  def get(url, query={})
+    self.class.get(url, query: query)
+  end
+
+  def get_all(url:, record_key:, limit: 100, query: {})
```
Comment (niquerio) [inline models/illiad_client.rb]: I think we should get rid of get_all because this logic is specific to the Alma API

**Review by niquerio — CHANGES_REQUESTED** [review]
I added the keys in kubernetes, so it should be good to go there. 

@erinesullivan can you remove the #get_all method in illiad_client.rb ? We can rewrite it to work with the ILLiad API when/if that's needed.

File: models/illiad_client.rb (hunk 2 above)
Comment (erinesullivan) [reply models/illiad_client.rb]: `get_all` has been removed.

**Review by niquerio — APPROVED** [review]

**Pull request merged**